packages/form-field/src/FormField.spec.tsx

```diff
@@ -453,47 +453,109 @@ describe('packages/form-field', () => {
     });
   });
 
-  test('Renders an icon', () => {
-    const { queryByTestId } = render(
-      <FormField label="Label" data-testid="form-field">
-        <FormFieldInputContainer
-          contentEnd={<Icon glyph="Beaker" data-testid="icon" />}
-        >
-          <div data-testid="input" />
-        </FormFieldInputContainer>
-      </FormField>,
-    );
+  describe('additional children', () => {
+    test('does not render if `optional` prop and `contentEnd` prop are undefined', () => {
+      const { queryByTestId } = render(
+        <FormField label="Label">
+          <FormFieldInputContainer>
+            <input />
+          </FormFieldInputContainer>
+        </FormField>,
+      );
+      expect(queryByTestId(LGIDS_FORM_FIELD.optional)).not.toBeInTheDocument();
+      expect(
+        queryByTestId(LGIDS_FORM_FIELD.contentEnd),
+      ).not.toBeInTheDocument();
+    });
 
-    const icon = queryByTestId('icon');
-    expect(icon).toBeInTheDocument();
-    expect(icon?.tagName.toLowerCase()).toEqual('svg');
-  });
+    describe('optional text', () => {
+      test(`renders if form field state is ${FormFieldState.None}, is not disabled, and optional prop is true`, () => {
+        const { getByTestId } = render(
+          <FormField label="Label" state={FormFieldState.None} optional>
+            <FormFieldInputContainer>
+              <input />
+            </FormFieldInputContainer>
+          </FormField>,
+        );
+        expect(getByTestId(LGIDS_FORM_FIELD.optional)).toBeInTheDocument();
+      });
 
-  test('Renders other content', () => {
-    const { queryByText } = render(
-      <FormField label="Label" data-testid="form-field">
-        <FormFieldInputContainer contentEnd={<em>Optional</em>}>
-          <div data-testid="input" />
-        </FormFieldInputContainer>
-      </FormField>,
-    );
+      test.each([FormFieldState.Error, FormFieldState.Valid])(
+        'does not render if form field state is %s',
```

Comment:
really?

Reply:
this was existing logic. let's chat about it monday if it's unexpected or no longer needed